derive/src/lib.rs

```diff
@@ -245,39 +260,45 @@ fn generate_variant_type(data_enum: &DataEnum, scale_info: &Ident) -> TokenStrea
         .filter(|v| !utils::should_skip(&v.attrs))
         .map(|v| {
             let ident = &v.ident;
-            let v_name = quote! {stringify!(#ident) };
-            match v.fields {
+            let v_name = quote! {::core::stringify!(#ident) };
+            let docs = utils::get_doc_literals(&v.attrs);
+            let index = utils::maybe_index(v).map(|i| quote!(.index(#i)));
```

Comment:
It's hard to overstate how much nicer this is compared to what I had.

Reply:
That was one of my goals with the builder pattern changes, so thanks!, The main priorities being type safe construction and friendliness for the derive macro. The tradeoff being that it is not fantastic for constructing type infos by hand, but we are assuming that is not very common.